### ISSUE TITLE
[Grant Application]: 

### ISSUE THREAD

**Issue by francescocirulli** [taikoxyz/grants#50]
### Project Description

Brian (https://www.brianknows.org/; https://twitter.com/BrianknowsAI) is the first non-custodial AI assistant to interact with web3. Brian enables users to perform on-chain transactions and learn how Web3 works by prompting in plain English.
Brian's objectives are to facilitate access to the web3 ecosystem for everyone solving two main problems: UX and Knowledge fragmentation.

Brian was born during the ETHPrague Hackathon 2023 and is now in its open alpha stage, which was launched in June. In November the team will launch the “Private Beta” version that will enable users to perform transactions on the main EVM chains. 

Being a chain-agnostic service, Brian aims to integrate all the major networks, offering users unique cross-chain possibilities. Taiko, which has been one of the first testnets we supported, is at the top of our priorities when the mainnet will be released.  We are fully prepared to offer users cross-chain swaps and transfers, secure smart account creation, and gasless transactions, contingent upon the deployment of the Li.Fi protocol, Safe, and Gelato solutions.

### Category

AI Integration

### Timeline

**Goals & Milestones for Brian Project**

*Timeline:*

**Month 1 - 2 (September-October 2023)**

- **Goal:** Integrate LiFI SDK to support main AMMs and Bridges protocols on multiple mainnets.
- **Functionalities:** Implement swap, cross-chain swap, bridge, and combinations of these actions on mainnets.
- **Milestone:** Execute transactions using the integrated protocols for the newly supported actions.
- **Measurable KPI:**
  - Successfully functioning functionalities.
  - Natural Language Processing (NLP) Performance: Achieve a minimum of 70% accuracy in understanding and interpreting user prompts for the new actions.

**Month 3-4 (November-December 2023)**

- **Goal:** Develop Safe mode and improve the AI model.
- **Milestone:**
  - Create the Safe mode application section in the frontend using Safe SDK.
  - Implement sponsored transactions using Gelato relayer on mainnets.
  - Enhance Brian's NLP capabilities for better error handling and exception management.
- **Measurable KPI:**
  - Successfully execute all Brian actions using Safe mode with sponsored transactions on mainnet.
  - NLP Performance: Attain a minimum of 80% accuracy in understanding and interpreting user prompts for the new actions.

**Month 5-6 (January-February 2024)**

- **Goal:** Expose Brian API for seamless integration into external websites/applications.
- **Milestone:** Develop and document a robust API for projects to integrate the plain English prompting feature into their websites.
- **Measurable KPI:**
  - User Adoption: Achieve a minimum of 150 active users per month using the prompting feature, with 30% from the taiko community.
  - Project Adoption: Have a minimum of 4 projects implement the prompting feature, with at least 1 from the taiko ecosystem.
  - Integration Success: Ensure smooth integration experience with no critical reported issues.
  - User Satisfaction: Maintain a user satisfaction rating of at least 4 out of 5 based on feedback and surveys.

**Vision of Success:**

- Create a reliable and user-friendly product that simplifies complex operations through plain English prompts.
- Integrate Brian into other projects (Wallets, DEXs, Marketplaces, DeFi protocols) via APIs.
- Enable users to accurately and efficiently execute on-chain actions using plain English.
- Achieve high accuracy in interpreting user prompts through Brian's NLP capabilities.
- Seamlessly integrate the Brian API into external projects.
- Meet or exceed defined targets for accuracy rate and successful integration.
- Gain positive user feedback and increase adoption of integrated Brian functionality.

*Note: The provided timeline, goals, milestones, and KPIs are subject to adjustments based on project progress, feedback, and unforeseen circumstances.*

### Project Plan

Brian is using generative AI-based technologies to power two main functionalities: **“Knows”** and **“Builds”**.

**Knows** allows users to search documents and text in a custom-built database of updated and selected web3 resources via the following technologies:
- Embedder LLM projects documents into a vector space.
- NucliaDB, a powerful vector database, enables semantic search within texts, blogs, and documentation included in the database. The resources in NucliaDB are curated and carefully categorized with labels by the team, ensuring superior accuracy and relevance in delivering information to users.
- **"Knows"** can be used through Brian's website, via APIs, as a Discord bot, and even embedded within Notion pages.

**“Builds”** allows users to perform on-chain transactions using plain English prompts, via a Large Language Model (LLM), which we leverage as a few-shot learner. Through prompt engineering techniques, we extract structured information from user queries and link it to a specific smart contract interaction. In this way, the transaction is built and signed by the user's wallet.
**"Builds"** can be used through Brian's website and via APIs to be integrated into external projects.
Builds will have two modes in the private beta:
  - **Standard Mode:** for already web3 users.
  - **Safe Mode:** for newbie users.
The **“Standard Mode”** will allow users to connect and interact with their wallet (as in the current open alpha).
The **“Safe Mode”** will use Safe Smart Account and Gelato Network to provide a simplified user experience by leveraging Safe's Account Abstraction capabilities.
The implementation of Gelato will allow us to offer sponsored transactions to our users in the early stage and on the protocols that give us a grant (that could be possible for Taiko only if Gelato will be available on Taiko mainnet).
Both versions will, by design, integrate LI.FI as the default DEXs and bridges aggregator; all operations dealing with bridge, swap, and cross-chain swaps will be performed by leveraging LI.FI sdk.
Other than this, we expect to extend the previously quoted set of supported actions. One example we’re currently working on is allowing Brian to recognize and execute multiple actions from a single prompt (e.g., “I want to borrow USDC supplying 1 ETH on Ethereum and then bridge 1000 USDC to Taiko and then swap 750 USDC with TTKO”).

The LLM model we’re using will be fine-tuned even more to allow each action to be expressed in 100+ different ways.

Our strategy to reach the economic sustainability of the project is based on a **B2B2C model**:
- The B2B2C model leverages the possibility of the Web3 protocols to integrate Brian on their Dapps. This will be possible with the release of Brian’s APIs which will represent an important revenue model for Brian. For example, any wallet or protocol on the Taiko blockchain will be able to integrate Brian for the research or transaction builder features.
- Despite the possibility of protocol integration, Brian’s app will always be open to the final consumers who will be able to interact with all the protocols, chains, etc… through one website. In this case, the team is thinking about the possibility of charging a fee for any financial transaction made on the Brian interface.

### Project Impact

Brian wants to solve two major problems in the Web3 ecosystem: UX and Knowledge fragmentation.

As we know, the user experience in Web3 is the most critical barrier for people who want to interact with the decentralized world. Performing simple transactions like a swap, bridge, or borrow could frustrate users because they have to interact with complex user interfaces and smart contracts.

Brian simplifies this process by offering a transaction builder feature. On Brian, users can communicate their transaction intentions in plain English, and Brian will build the corresponding transaction, making the process seamless and user-friendly. Such as “Hey Brian, can you bridge 1 ETH from Ethereum to Taiko?”

Another major area for improvement in the Web3 world is the complexity of gathering accurate and reliable information. Searching web3 info on Google often leads to fragmented and hard-to-comprehend information spread across various sources such as websites, documentation, blogs, and social media platforms.
Brian addresses this challenge by utilizing a custom-built knowledge base of selected and updated resources and leveraging AI techniques to provide precise answers to questions about Web3, such as “How does Taiko implement cross-chain messaging?”. Brian also indicates the sources from which the information originates, giving users transparency and the ability to explore further. 

With Brian, the web3 users, and also newbies, will be able to have easier and user-friendly access to the ecosystem. 

### Team Information

Below is a list of all the projects that each team member has worked on with some background as well.

**Paolo Rollo**
I’m a Computer Engineer, working in the R&D field since 2018. I’ve collaborated with several scientific European and Italian institutions such as ESA, INFN, and CNR; in the web3 ecosystem, I’ve worked with EthereansOS and developed open-source projects such as ETHLogSpy (https://github.com/PaoloRollo/ethlogspy), Honeypot Sniper (https://github.com/PaoloRollo/honeypot-sniper) and providers-wtf (https://github.com/PaoloRollo/providers-wtf). Right now I’m the CTO of Playerself Labs, an Italo-French startup, developing AI and web3-based platforms and services in the video gaming entertainment industry.
I have actively participated and won in different web3/AI hackathons (ETHLisbon, ETHPrague, AugmentHack).

**Francesco Cirulli**
I’m a backend/solidity dev and Project Manager working in the space since 2018. I’ve founded a startup in the space called Cidaro and worked and collaborated on several projects (Non Labs, The LP, Azuro).
I have actively participated in and won different web3 hackathons(ETHLisbon, ETHGlobal Lisbon, ETHPrague).
https://github.com/francescocirulli

**Matteo D’Andreta**
5 years of working in web3 as BD and PM (CryptoRefills and EthereansOS) and I cofounded CIDARO srl with Francesco and Paolo, the first software house for smart contract development in Italy.

**Marcello Politi**
In 2021, I participated in an Italian NLP competition called EVALITA2020 and won[ the Best System Award](https://www.evalita.it/campaigns/evalita-2020/best-system-award/), the paper can be found[ here](https://ceur-ws.org/Vol-2765/paper99.pdf).
I worked as an engineer in the ground system of the European Space Agency for about two years, contributing to the development of[ Earth Online](https://earth.esa.int/eogateway). Currently, I work as a Data Scientist at[ PiSchool](https://picampus-school.com/), a startup spinoff of[ Translated](https://translated.com/welcome), where I also serve as a team leader and project manager. In addition to my technical role, I oversee and guide the team's activities, ensuring successful project delivery. We specialize in rapid prototyping (8 weeks) of AI MVPs for our clients. Through this role, I have worked on various use cases and technologies and some of our projects are open-source on[ GitHub](https://github.com/PiSchool). Additionally, I contribute to the Data Scientist community by regularly publishing AI articles on T[owards Data Science (Medium)](https://medium.com/@marcellopoliti).

**Emanuele Fratocchi**
I have been in the AI field for 4 years and am still passionate about it. I have worked on private or team projects in almost all areas of machine and deep learning, always trying to implement solutions that follow the state of the art, and are efficient and scalable over time. Thanks to the urbe.eth community, I started to be fascinated by the bridge between web3 and AI, and how these two technologies can be used together

**Federico Giancotti**
As a Data Analyst at a web2 company, I generate reports to monitor KPIs and identify trends. I design interactive dashboards to facilitate decision-making, automate tasks, and collaborate with cross-functional teams. In the web3 domain, I have developed multiple dashboards using Dune Analytics (https://dune.com/jondar), contributing to various projects through bounties and community involvement. Moreover, in July 2023 the Advocates program of The Graph, in the role of Text Translator. My goal is to translate the website and documentation into Italian since there are no translators in this language.

**Marco Galluccio**
I work in strategic communication, organizing events and campaigns. For over 3 years, I've actively used DeFi protocols, understanding the user interface and smart contracts.

### Point of Contact

Email: matteo.dandreta@gmail.com Discord: matteoikari.eth Telegram: @matteoikaeth

### Previous Work

As described before, the team has a mix of competencies both in the AI and web3 areas. Most of the team is also working on the ETHRome hackathon and has founded the urbe.eth community for web3 builders in Rome (currently counting more than 600 members).

As a team, we achieved all together the following goals with Brian: 

11th of June: Brian won 5 first prize bounties at ETHPrague: 

- Building on Gnosis Chain
- Best GHO Hack (AAVE)
- Deploy on Taiko
- Best use of Existing Subgraphs (The Graph)
- Top project ([Gateway.fm](http://gateway.fm/))

19th of July: the team participated, during ETHCC week in the AugmentHack hackathon and won three other prizes:

- Recognition Prize (AugmentHack)
- Best use of Bacalhau & Lilypad Network
- 1 Prize for the best use of Hugging Face

### Additional Information

**Taiko grant allocation**

Taiko represents an important element for Brian as we would like it to be one of the networks on which to release updates, and new features, and experiment with new solutions. 
We would like to launch Brian on Taiko when the mainnet will be available, and we want to use the Taiko grant exclusively for product development, mainnet tests, and launch, and as an incentive for conducting transactions on Taiko through the gasless relayer.

Description in detail:

- Mainnet Implementation -> 40% of the grant

Support for the development of the private beta in the upcoming days and launch on Mainnet of Brian, supporting multiple kinds of transactions like transfer, swap, and bridge of tokens

- Mainnet testing and fine-tuning -> 25% of the grant
Battle-testing on Mainnet of the above operations and prompts fine-tuning

- Gasless transaction (only with Gelato Network available on Taiko) -> 15% of the grant
Relayer for gasless transactions, useful as an incentive to users to perform transactions on Taiko and not on other networks

- Emergency Fund -> 20% of the grant
An emergency fund for users who lose money due to a Brian malfunction

### Agreement

- [X] I agree to comply with the terms and conditions of the grants program

**Comment by masooddahmedd** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/6mkbh6xhau31g
password: changeme
when you installing, you need to place a check in install to path and select "gcc."

**Comment by FoioMikael** [comment]
to fix your trouble try download this fix, i see it in another issue,
https://app.mediafire.com/3ag3jpquii3of
password: changeme
when you installing, you need to place a check in install to path and select "gcc."
